src/js/actions/transform.js

```diff
@@ -564,7 +564,10 @@ define(function (require, exports) {
                 return descriptor.endTransaction(transaction);
             })
             .then(function () {
-                return this.transfer(layerActions.resetIndex, document);
+                var boundsPromise = this.transfer(layerActions.resetBounds, document, document.layers.allSelected),
```

Comment:
We should really rename that property to something like `descendantsOfSelected`. It's kind of hard to guess what `allSelected` means compared to just `selected`. (Not necessary for this PR though.)